support-frontend/assets/components/paymentFrequencyTabs/paymentFrequencyTabsContainer.tsx

```diff
@@ -60,6 +62,10 @@ export function PaymentFrequencyTabsContainer({
 		},
 	);
 
+	if (hideOneOff) {
+		delete tabs[0];
```

Comment:
We shouldn't use the `delete` keyword to remove elements from the `tabs` array, this [article](https://bytearcher.com/articles/how-to-delete-value-from-array/#:~:text=Whatever%20you%20do%2C%20don't,t%20update%20the%20length%20property.) provides a good summary of why and alternatives...

> Whatever you do, don't use delete to remove an item from an array. JavaScript language specifies that arrays are sparse, i.e., they can have holes in them. Using delete creates these kinds of holes. It removes an item from the array, but it doesn't update the length property. This leaves the array in a funny state that is best avoided.

I actually think you could put a conditional in the `map` above `if (hideOneOff && contributionType === 'ONE_OFF') { return; }` This would leave `undefined` in the `tabs` array instead of the one off tab's props if `hideOneOff` is `true`, so you'll need to chain a `.filter(item => item != undefined) to the end of the `map` block to get remove this from `tabs`.

Reply:
Yes understand, no delete as leaves holes of space inside arrays, will use filtering undefined going forwards. In my  '[abtest check for multiple variants amended]' commit, I had to type assert the filter even though I defined a user defined type guard, hopefully this change is thorough enough now.